File: Server/src/infrastructure/database/models/Partner.js
module.exports = (sequelize, DataTypes) => {
    const Partner = sequelize.define("Partner", {
        partner_id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        name: DataTypes.STRING,
        address: DataTypes.TEXT,
    });

    Partner.associate = (models) => {
        Partner.hasMany(models.PartnerStock, {
            foreignKey: "partner_id",
            as: "PartnerStocks",
        });
        Partner.hasMany(models.Sale, {
            foreignKey: "partner_id",
            as: "Sales",
        });
    };

    return Partner;
};

File: app/src/main/java/com/praktikum/honeypot/Data/Partner.kt
package com.praktikum.honeypot.Data

data class Partner(
    val partner_id: Int,
    val name: String,
    val address: String,
    val imageUrl: String?,  // New attribute for partner image URL
    val PartnerStocks: List<PartnerStock> = emptyList()
)

